crud.py

```diff
@@ -0,0 +1,67 @@
+from sqlalchemy.orm import Session
+
+from db import models, schemas
+from db.models import DBAuthor, DBBook
+
+
+def get_author_by_name(
+    db: Session,
+    name: str,
+) -> DBAuthor | None:
+    return db.query(models.DBAuthor).filter(models.DBAuthor.name == name).first()
+
+
+def get_all_authors(db: Session, page: int = 0, page_size: int = 20) -> list[DBAuthor]:
+    return db.query(models.DBAuthor).offset(page).limit(page_size).all()
+
+
+def get_detailed_author(db: Session, author_id: int = None) -> DBAuthor | None:
+    return db.query(models.DBAuthor).filter(models.DBAuthor.id == author_id).first()
+
+
+def create_author(db: Session, author: schemas.AuthorCreate) -> DBAuthor:
+    db_author = models.DBAuthor(
+        name=author.name,
+    )
+    db.add(db_author)
+    db.commit()
+    db.refresh(db_author)
+
+    return db_author
+
+
+def get_all_books(
+    db: Session, author_id: int = None, page: int = 0, page_size: int = 20
```

Comment:
```suggestion
    db: Session, author_id: Optional[int] = None, page: int = 0, page_size: int = 20
```
Because none is default parameter